diff --git a/dist/index.js b/dist/index.js
index d1bb042b86105ee3145bd5264c7a769be74f67d4..bac2ed18a23b0a716aabccd05633f9b9fd2f272a 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -2334,7 +2334,10 @@ var Microcontroller_RP2040 = (_a) => {
         schSectionName: schSections.power,
         pcbX: -2,
         pcbY: 21,
-        pcbRotation: 90
+        pcbRotation: 90,
+        schX: 7.8,
+        schY: -4.6,
+        schRotation: 180
       }
     ),
     /* @__PURE__ */ jsx20("trace", __spreadValues({ name: "VBUS_D", from: "net.VBUS", to: ".D_VBUS > .anode" }, vbusLabel)),
@@ -2355,7 +2358,10 @@ var Microcontroller_RP2040 = (_a) => {
         schSectionName: schSections.power,
         pcbX: -4.7,
         pcbY: 17.3,
-        pcbRotation: 90
+        pcbRotation: 90,
+        schOrientation: "horizontal",
+        schX: 9.4,
+        schY: -4.6
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2379,7 +2385,7 @@ var Microcontroller_RP2040 = (_a) => {
         pcbX: -6,
         pcbY: -10,
         schX: -9.6,
-        schY: 0
+        schY: -0.5
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2393,7 +2399,7 @@ var Microcontroller_RP2040 = (_a) => {
         pcbX: -8,
         pcbY: 1.6,
         schX: -7.2,
-        schY: 2.31
+        schY: 1
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2426,7 +2432,7 @@ var Microcontroller_RP2040 = (_a) => {
         pcbX: -8,
         pcbY: -1,
         schX: -7.2,
-        schY: -2.31
+        schY: -2
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2440,7 +2446,7 @@ var Microcontroller_RP2040 = (_a) => {
         pcbX: -4,
         pcbY: -6,
         schX: -7.9,
-        schY: 0
+        schY: -0.5
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2469,7 +2475,9 @@ var Microcontroller_RP2040 = (_a) => {
         footprint: "0402",
         schSectionName: schSections.controls,
         pcbX: 10.4,
-        pcbY: -5.5
+        pcbY: -5.5,
+        schX: 12.5,
+        schY: -11
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2498,7 +2506,7 @@ var Microcontroller_RP2040 = (_a) => {
         pcbX: -4,
         pcbY: -5,
         schX: -10.8,
-        schY: -2.31
+        schY: -2
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2512,7 +2520,7 @@ var Microcontroller_RP2040 = (_a) => {
         pcbX: -3,
         pcbY: 6,
         schX: -10.8,
-        schY: 2.31
+        schY: 1
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2526,7 +2534,7 @@ var Microcontroller_RP2040 = (_a) => {
         pcbX: 7.2,
         pcbY: 14.5,
         schX: 16.41,
-        schY: 1.2
+        schY: 0.5
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2538,7 +2546,9 @@ var Microcontroller_RP2040 = (_a) => {
         schSectionName: schSections.power,
         schOrientation: "vertical",
         pcbX: 8,
-        pcbY: -4.4
+        pcbY: -4.4,
+        schX: 6.5,
+        schY: -6.3
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2603,7 +2613,15 @@ var Microcontroller_RP2040 = (_a) => {
         schSectionName: schSections.usb,
         pcbX: 0,
         pcbY: 31,
-        pcbRotation: 180
+        pcbRotation: 180,
+        schX: 12.25,
+        schY: -5.8,
+        schWidth: 2.15,
+        schHeight: 1.8,
+        schPinArrangement: {
+          leftSide: [13, 15, 17, 18, 20, 22, 23, 25],
+          rightSide: [14, 16, 28, 27, 26, 24, 21, 19]
+        }
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2616,7 +2634,7 @@ var Microcontroller_RP2040 = (_a) => {
         pcbX: 0,
         pcbY: 0.5,
         schX: -0.22,
-        schY: -1,
+        schY: -2,
         schWidth: 2.8,
         schHeight: 5.8
       }
@@ -2629,9 +2647,12 @@ var Microcontroller_RP2040 = (_a) => {
         pcbX: 3.4,
         pcbY: 9.5,
         pcbRotation: 90,
-        schX: 18.02,
+        schX: 17.5,
         schY: -4.16,
-        schHeight: 1
+        schHeight: 1.6,
+        schPinArrangement: {
+          leftSide: [8, 1, 2, 3, 5, 6, 7, 4, 9]
+        }
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2642,6 +2663,8 @@ var Microcontroller_RP2040 = (_a) => {
         pcbX: -7.2,
         pcbY: 20.2,
         pcbRotation: 180,
+        schX: 8.8,
+        schY: -6.3,
         schHeight: 0.6
       }
     ),
@@ -2651,7 +2674,9 @@ var Microcontroller_RP2040 = (_a) => {
         name: "Y1",
         schSectionName: schSections.clock,
         pcbX: -0.5,
-        pcbY: -6
+        pcbY: -6,
+        schX: 1.025,
+        schY: -15.2
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2660,7 +2685,9 @@ var Microcontroller_RP2040 = (_a) => {
         name: "SW_BOOT",
         schSectionName: schSections.controls,
         pcbX: 8.6,
-        pcbY: 21.8
+        pcbY: 21.8,
+        schX: 10.6,
+        schY: -9.4
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2670,7 +2697,9 @@ var Microcontroller_RP2040 = (_a) => {
         schSectionName: schSections.controls,
         pcbX: 5.5,
         pcbY: -12.5,
-        pcbRotation: 90
+        pcbRotation: 90,
+        schX: 12.5,
+        schY: -9.4
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2681,7 +2710,9 @@ var Microcontroller_RP2040 = (_a) => {
         schSectionName: schSections.status,
         pcbX: 10,
         pcbY: 4.2,
-        pcbRotation: 90
+        pcbRotation: 90,
+        schX: 9.66,
+        schY: -13.66
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2694,7 +2725,7 @@ var Microcontroller_RP2040 = (_a) => {
         pcbY: 24.8,
         pcbRotation: 90,
         schX: 10.2,
-        schY: -18.52
+        schY: -15.5
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2706,7 +2737,9 @@ var Microcontroller_RP2040 = (_a) => {
         schSectionName: schSections.controls,
         pcbX: 12,
         pcbY: 17.8,
-        pcbRotation: 90
+        pcbRotation: 90,
+        schX: 10.6,
+        schY: -11
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2718,7 +2751,9 @@ var Microcontroller_RP2040 = (_a) => {
         schSectionName: schSections.status,
         pcbX: 7.2,
         pcbY: 1.2,
-        pcbRotation: 90
+        pcbRotation: 90,
+        schX: 8.06,
+        schY: -13.66
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2733,7 +2768,7 @@ var Microcontroller_RP2040 = (_a) => {
         pcbRotation: 90,
         schOrientation: "horizontal",
         schX: 8.6,
-        schY: -18.52
+        schY: -15.5
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2744,7 +2779,9 @@ var Microcontroller_RP2040 = (_a) => {
         footprint: "0402",
         schSectionName: schSections.usb,
         pcbX: -0.2,
-        pcbY: 25.6
+        pcbY: 25.6,
+        schX: 8.89,
+        schY: -7.36
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2770,7 +2807,7 @@ var Microcontroller_RP2040 = (_a) => {
         pcbX: 3.4,
         pcbY: 22.5,
         pcbRotation: 90,
-        schX: 14.96,
+        schX: 15.3,
         schY: -6.1
       }
     ),
@@ -2799,8 +2836,8 @@ var Microcontroller_RP2040 = (_a) => {
         pcbX: -2.8,
         pcbY: 26.3,
         pcbRotation: 90,
-        schX: 15.47,
-        schY: -3.96
+        schX: 14.7,
+        schY: -3
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2812,7 +2849,9 @@ var Microcontroller_RP2040 = (_a) => {
         schSectionName: schSections.power,
         schOrientation: "vertical",
         pcbX: -8.5,
-        pcbY: 4.2
+        pcbY: 4.2,
+        schX: 9.2,
+        schY: -8.3
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2852,7 +2891,9 @@ var Microcontroller_RP2040 = (_a) => {
         schSectionName: schSections.clock,
         schOrientation: "vertical",
         pcbX: -8.4,
-        pcbY: -11.8
+        pcbY: -11.8,
+        schX: -0.26,
+        schY: -17
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2864,7 +2905,9 @@ var Microcontroller_RP2040 = (_a) => {
         schSectionName: schSections.clock,
         schOrientation: "vertical",
         pcbX: -2.4,
-        pcbY: -21
+        pcbY: -21,
+        schX: 1.565,
+        schY: -17
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2877,7 +2920,9 @@ var Microcontroller_RP2040 = (_a) => {
         pcbX: 8.5,
         pcbY: -1.8,
         supplierPartNumbers: { jlcpcb: ["C1002"] },
-        pcbRotation: 90
+        pcbRotation: 90,
+        schX: 6.8,
+        schY: -8.3
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2889,7 +2934,9 @@ var Microcontroller_RP2040 = (_a) => {
         padDiameter: "1.1mm",
         schSectionName: schSections.debug,
         pcbX: -6,
-        pcbY: -31
+        pcbY: -31,
+        schX: 6,
+        schY: -18.1
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2901,7 +2948,9 @@ var Microcontroller_RP2040 = (_a) => {
         padDiameter: "1.1mm",
         schSectionName: schSections.debug,
         pcbX: -2,
-        pcbY: -31
+        pcbY: -31,
+        schX: 7.5,
+        schY: -18.1
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2913,7 +2962,9 @@ var Microcontroller_RP2040 = (_a) => {
         padDiameter: "1.1mm",
         schSectionName: schSections.debug,
         pcbX: 2,
-        pcbY: -31
+        pcbY: -31,
+        schX: 9,
+        schY: -18.1
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2925,7 +2976,9 @@ var Microcontroller_RP2040 = (_a) => {
         padDiameter: "1.1mm",
         schSectionName: schSections.debug,
         pcbX: 6,
-        pcbY: -31
+        pcbY: -31,
+        schX: 10.5,
+        schY: -18.1
       }
     ),
     /* @__PURE__ */ jsx20(
