diff --git a/dist/index.js b/dist/index.js
index 8cca3643e716c563e29084ea419c9cab5e2ba833..005d67992a795530535eb38a3bd8130000a3d6f3 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -2304,15 +2304,15 @@ var adcRefLabel = {
   schDisplayLabel: "ADC_REF"
 };
 var schSections = {
-  rp2040: "rp2040",
-  usb: "usb",
-  power: "power",
-  flash: "flash",
-  clock: "clock",
-  controls: "controls",
-  display: "display",
-  status: "status",
-  debug: "debug"
+  rp2040: (name) => `${name}__rp2040`,
+  usb: (name) => `${name}__usb`,
+  power: (name) => `${name}__power`,
+  flash: (name) => `${name}__flash`,
+  clock: (name) => `${name}__clock`,
+  controls: (name) => `${name}__controls`,
+  display: (name) => `${name}__display`,
+  status: (name) => `${name}__status`,
+  debug: (name) => `${name}__debug`
 };
 var Microcontroller_RP2040 = (_a) => {
   var _b = _a, {
@@ -2323,6 +2323,28 @@ var Microcontroller_RP2040 = (_a) => {
     "connections"
   ]);
   return /* @__PURE__ */ jsxs11("subcircuit", __spreadProps(__spreadValues({ name }, props), { children: [
+    /* @__PURE__ */ jsx20(
+      "schematicsection",
+      {
+        name: schSections.rp2040(name),
+        displayName: "RP2040 & Power"
+      }
+    ),
+    /* @__PURE__ */ jsx20(
+      "schematicsection",
+      {
+        name: schSections.usb(name),
+        displayName: "Programming USB-C & QSPI"
+      }
+    ),
+    /* @__PURE__ */ jsx20("schematicsection", { name: schSections.clock(name), displayName: "Clock" }),
+    /* @__PURE__ */ jsx20(
+      "schematicsection",
+      {
+        name: schSections.status(name),
+        displayName: "Status & SWD Debug"
+      }
+    ),
     /* @__PURE__ */ jsx20("trace", __spreadValues({ name: "Y1_G1", from: ".Y1 > .pin2", to: "net.GND" }, gndLabel)),
     /* @__PURE__ */ jsx20("trace", __spreadValues({ name: "Y1_G2", from: ".Y1 > .pin4", to: "net.GND" }, gndLabel)),
     /* @__PURE__ */ jsx20("trace", { name: "USB_DN_B", from: ".J_USB > .B7", to: ".R_USB1 > .pin1" }),
@@ -2331,10 +2353,13 @@ var Microcontroller_RP2040 = (_a) => {
       B5819W_SL,
       {
         name: "D_VBUS",
-        schSectionName: schSections.power,
+        schSectionName: schSections.power(name),
         pcbX: -2,
         pcbY: 21,
-        pcbRotation: 90
+        pcbRotation: 90,
+        schX: 7.8,
+        schY: -4.6,
+        schRotation: 180
       }
     ),
     /* @__PURE__ */ jsx20("trace", __spreadValues({ name: "VBUS_D", from: "net.VBUS", to: ".D_VBUS > .anode" }, vbusLabel)),
@@ -2352,10 +2377,13 @@ var Microcontroller_RP2040 = (_a) => {
         name: "R_3V3_EN",
         resistance: "100k",
         footprint: "0402",
-        schSectionName: schSections.power,
+        schSectionName: schSections.power(name),
         pcbX: -4.7,
         pcbY: 17.3,
-        pcbRotation: 90
+        pcbRotation: 90,
+        schOrientation: "horizontal",
+        schX: 9.4,
+        schY: -4.6
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2367,19 +2395,76 @@ var Microcontroller_RP2040 = (_a) => {
       }, vsysLabel)
     ),
     /* @__PURE__ */ jsx20("trace", { name: "EN_R", from: ".R_3V3_EN > .pin2", to: ".U3 > .EN" }),
+    /* @__PURE__ */ jsx20(
+      "capacitor",
+      {
+        name: "C_IOVDD1",
+        capacitance: "100nF",
+        footprint: "0402",
+        schSectionName: schSections.rp2040(name),
+        schOrientation: "vertical",
+        pcbX: -4,
+        pcbY: -5,
+        schX: -8.15,
+        schY: -0.86
+      }
+    ),
+    /* @__PURE__ */ jsx20(
+      "capacitor",
+      {
+        name: "C_IOVDD2",
+        capacitance: "100nF",
+        footprint: "0402",
+        schSectionName: schSections.rp2040(name),
+        schOrientation: "vertical",
+        pcbX: -3,
+        pcbY: 6,
+        schX: -6.225,
+        schY: -0.86
+      }
+    ),
+    /* @__PURE__ */ jsx20(
+      "capacitor",
+      {
+        name: "C_IOVDD3",
+        capacitance: "100nF",
+        footprint: "0402",
+        schSectionName: schSections.rp2040(name),
+        schOrientation: "vertical",
+        pcbRotation: -90,
+        pcbX: -8,
+        pcbY: -1,
+        schX: -4.3,
+        schY: -0.86
+      }
+    ),
+    /* @__PURE__ */ jsx20(
+      "capacitor",
+      {
+        name: "C_IOVDD4",
+        capacitance: "100nF",
+        footprint: "0402",
+        schSectionName: schSections.rp2040(name),
+        schOrientation: "vertical",
+        pcbX: -4,
+        pcbY: -6,
+        schX: -8.15,
+        schY: -2.46
+      }
+    ),
     /* @__PURE__ */ jsx20(
       "capacitor",
       {
         name: "C_IOVDD5",
         capacitance: "100nF",
         footprint: "0402",
-        schSectionName: schSections.rp2040,
+        schSectionName: schSections.rp2040(name),
         schOrientation: "vertical",
         pcbRotation: -90,
         pcbX: -6,
         pcbY: -10,
-        schX: -9.6,
-        schY: 0
+        schX: -6.225,
+        schY: -2.46
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2388,12 +2473,12 @@ var Microcontroller_RP2040 = (_a) => {
         name: "C_IOVDD6",
         capacitance: "100nF",
         footprint: "0402",
-        schSectionName: schSections.rp2040,
+        schSectionName: schSections.rp2040(name),
         schOrientation: "vertical",
         pcbX: -8,
         pcbY: 1.6,
-        schX: -7.2,
-        schY: 2.31
+        schX: -4.3,
+        schY: -2.46
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2414,35 +2499,6 @@ var Microcontroller_RP2040 = (_a) => {
       }, v3v3Label)
     ),
     /* @__PURE__ */ jsx20("trace", __spreadValues({ name: "IO6_G", from: ".C_IOVDD6 > .pin2", to: "net.GND" }, gndLabel)),
-    /* @__PURE__ */ jsx20(
-      "capacitor",
-      {
-        name: "C_IOVDD3",
-        capacitance: "100nF",
-        footprint: "0402",
-        schSectionName: schSections.rp2040,
-        schOrientation: "vertical",
-        pcbRotation: -90,
-        pcbX: -8,
-        pcbY: -1,
-        schX: -7.2,
-        schY: -2.31
-      }
-    ),
-    /* @__PURE__ */ jsx20(
-      "capacitor",
-      {
-        name: "C_IOVDD4",
-        capacitance: "100nF",
-        footprint: "0402",
-        schSectionName: schSections.rp2040,
-        schOrientation: "vertical",
-        pcbX: -4,
-        pcbY: -6,
-        schX: -7.9,
-        schY: 0
-      }
-    ),
     /* @__PURE__ */ jsx20(
       "trace",
       __spreadValues({
@@ -2467,9 +2523,11 @@ var Microcontroller_RP2040 = (_a) => {
         name: "R_RUN",
         resistance: "10k",
         footprint: "0402",
-        schSectionName: schSections.controls,
+        schSectionName: schSections.controls(name),
         pcbX: 10.4,
-        pcbY: -5.5
+        pcbY: -5.5,
+        schX: 12.5,
+        schY: -11
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2478,41 +2536,11 @@ var Microcontroller_RP2040 = (_a) => {
         name: "C_FLASH",
         capacitance: "100nF",
         footprint: "0402",
-        schSectionName: schSections.flash,
+        schSectionName: schSections.flash(name),
         schOrientation: "vertical",
         pcbX: -5.5,
         pcbY: 10.8,
-        pcbRotation: 180,
-        schX: 16.65,
-        schY: -6.92
-      }
-    ),
-    /* @__PURE__ */ jsx20(
-      "capacitor",
-      {
-        name: "C_IOVDD1",
-        capacitance: "100nF",
-        footprint: "0402",
-        schSectionName: schSections.rp2040,
-        schOrientation: "vertical",
-        pcbX: -4,
-        pcbY: -5,
-        schX: -10.8,
-        schY: -2.31
-      }
-    ),
-    /* @__PURE__ */ jsx20(
-      "capacitor",
-      {
-        name: "C_IOVDD2",
-        capacitance: "100nF",
-        footprint: "0402",
-        schSectionName: schSections.rp2040,
-        schOrientation: "vertical",
-        pcbX: -3,
-        pcbY: 6,
-        schX: -10.8,
-        schY: 2.31
+        pcbRotation: 180
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2521,12 +2549,10 @@ var Microcontroller_RP2040 = (_a) => {
         name: "C_USB_VDD",
         capacitance: "100nF",
         footprint: "0402",
-        schSectionName: schSections.usb,
+        schSectionName: schSections.usb(name),
         schOrientation: "vertical",
         pcbX: 7.2,
-        pcbY: 14.5,
-        schX: 16.41,
-        schY: 1.2
+        pcbY: 14.5
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2535,7 +2561,7 @@ var Microcontroller_RP2040 = (_a) => {
         name: "C_ADC",
         capacitance: "100nF",
         footprint: "0402",
-        schSectionName: schSections.power,
+        schSectionName: schSections.power(name),
         schOrientation: "vertical",
         pcbX: 8,
         pcbY: -4.4
@@ -2600,10 +2626,18 @@ var Microcontroller_RP2040 = (_a) => {
       TYPE_C_16PIN_2MD_073_,
       {
         name: "J_USB",
-        schSectionName: schSections.usb,
+        schSectionName: schSections.usb(name),
         pcbX: 0,
         pcbY: 31,
-        pcbRotation: 180
+        pcbRotation: 180,
+        schX: 12.25,
+        schY: -5.8,
+        schWidth: 2.15,
+        schHeight: 1.8,
+        schPinArrangement: {
+          leftSide: [13, 15, 17, 18, 20, 22, 23, 25],
+          rightSide: [14, 16, 28, 27, 26, 24, 21, 19]
+        }
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2612,11 +2646,11 @@ var Microcontroller_RP2040 = (_a) => {
         name: "U1",
         connections,
         showPinAliases: true,
-        schSectionName: schSections.rp2040,
+        schSectionName: schSections.rp2040(name),
         pcbX: 0,
         pcbY: 0.5,
-        schX: -0.22,
-        schY: -1,
+        schX: -0.08,
+        schY: -2.5,
         schWidth: 2.8,
         schHeight: 5.8
       }
@@ -2625,23 +2659,28 @@ var Microcontroller_RP2040 = (_a) => {
       W25Q16JVUXIQ,
       {
         name: "U2",
-        schSectionName: schSections.flash,
+        schSectionName: schSections.flash(name),
         pcbX: 3.4,
         pcbY: 9.5,
         pcbRotation: 90,
-        schX: 18.02,
+        schX: 17.5,
         schY: -4.16,
-        schHeight: 1
+        schHeight: 1.6,
+        schPinArrangement: {
+          leftSide: [8, 1, 2, 3, 5, 6, 7, 4, 9]
+        }
       }
     ),
     /* @__PURE__ */ jsx20(
       AP2112K_3_3TRG1,
       {
         name: "U3",
-        schSectionName: schSections.power,
+        schSectionName: schSections.power(name),
         pcbX: -7.2,
         pcbY: 20.2,
         pcbRotation: 180,
+        schX: 8.8,
+        schY: -6.3,
         schHeight: 0.6
       }
     ),
@@ -2649,28 +2688,34 @@ var Microcontroller_RP2040 = (_a) => {
       X322512MSB4SI,
       {
         name: "Y1",
-        schSectionName: schSections.clock,
+        schSectionName: schSections.clock(name),
         pcbX: -0.5,
-        pcbY: -6
+        pcbY: -6,
+        schX: 1.025,
+        schY: -12.5
       }
     ),
     /* @__PURE__ */ jsx20(
       SKRPACE010,
       {
         name: "SW_BOOT",
-        schSectionName: schSections.controls,
+        schSectionName: schSections.controls(name),
         pcbX: 8.6,
-        pcbY: 21.8
+        pcbY: 21.8,
+        schX: 10.6,
+        schY: -9.4
       }
     ),
     /* @__PURE__ */ jsx20(
       SKRPACE010,
       {
         name: "SW_RUN",
-        schSectionName: schSections.controls,
+        schSectionName: schSections.controls(name),
         pcbX: 5.5,
         pcbY: -12.5,
-        pcbRotation: 90
+        pcbRotation: 90,
+        schX: 12.5,
+        schY: -9.4
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2678,10 +2723,12 @@ var Microcontroller_RP2040 = (_a) => {
       {
         name: "D1",
         color: "green",
-        schSectionName: schSections.status,
+        schSectionName: schSections.status(name),
         pcbX: 10,
         pcbY: 4.2,
-        pcbRotation: 90
+        pcbRotation: 90,
+        schX: 9.66,
+        schY: -12
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2689,12 +2736,12 @@ var Microcontroller_RP2040 = (_a) => {
       {
         name: "D_PWR",
         color: "green",
-        schSectionName: schSections.status,
+        schSectionName: schSections.status(name),
         pcbX: -9.8,
         pcbY: 24.8,
         pcbRotation: 90,
         schX: 10.2,
-        schY: -18.52
+        schY: -13.8
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2703,10 +2750,12 @@ var Microcontroller_RP2040 = (_a) => {
         name: "R_BOOT",
         resistance: "10k",
         footprint: "0402",
-        schSectionName: schSections.controls,
+        schSectionName: schSections.controls(name),
         pcbX: 12,
         pcbY: 17.8,
-        pcbRotation: 90
+        pcbRotation: 90,
+        schX: 10.6,
+        schY: -11
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2715,10 +2764,12 @@ var Microcontroller_RP2040 = (_a) => {
         name: "R_LED",
         resistance: "330",
         footprint: "0402",
-        schSectionName: schSections.status,
+        schSectionName: schSections.status(name),
         pcbX: 7.2,
         pcbY: 1.2,
-        pcbRotation: 90
+        pcbRotation: 90,
+        schX: 8.06,
+        schY: -12
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2727,13 +2778,13 @@ var Microcontroller_RP2040 = (_a) => {
         name: "R_PWR_LED",
         resistance: "330",
         footprint: "0402",
-        schSectionName: schSections.status,
+        schSectionName: schSections.status(name),
         pcbX: -6.2,
         pcbY: 24.8,
         pcbRotation: 90,
         schOrientation: "horizontal",
         schX: 8.6,
-        schY: -18.52
+        schY: -13.8
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2742,9 +2793,11 @@ var Microcontroller_RP2040 = (_a) => {
         name: "R_CC1",
         resistance: "5.1k",
         footprint: "0402",
-        schSectionName: schSections.usb,
+        schSectionName: schSections.usb(name),
         pcbX: -0.2,
-        pcbY: 25.6
+        pcbY: 25.6,
+        schX: 8.89,
+        schY: -7.36
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2753,11 +2806,9 @@ var Microcontroller_RP2040 = (_a) => {
         name: "R_CC2",
         resistance: "5.1k",
         footprint: "0402",
-        schSectionName: schSections.usb,
+        schSectionName: schSections.usb(name),
         pcbX: 3.6,
-        pcbY: 26.5,
-        schX: 13.91,
-        schY: -4.12
+        pcbY: 26.5
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2766,11 +2817,11 @@ var Microcontroller_RP2040 = (_a) => {
         name: "R_USB1",
         resistance: "27",
         footprint: "0402",
-        schSectionName: schSections.usb,
+        schSectionName: schSections.usb(name),
         pcbX: 3.4,
         pcbY: 22.5,
         pcbRotation: 90,
-        schX: 14.96,
+        schX: 15.3,
         schY: -6.1
       }
     ),
@@ -2780,7 +2831,7 @@ var Microcontroller_RP2040 = (_a) => {
         name: "R_USB2",
         resistance: "27",
         footprint: "0402",
-        schSectionName: schSections.usb,
+        schSectionName: schSections.usb(name),
         pcbX: 2.4,
         pcbY: 16.5,
         pcbRotation: -90,
@@ -2794,13 +2845,11 @@ var Microcontroller_RP2040 = (_a) => {
         name: "C_VBUS",
         capacitance: "10uF",
         footprint: "0603",
-        schSectionName: schSections.usb,
+        schSectionName: schSections.usb(name),
         schOrientation: "vertical",
         pcbX: -2.8,
         pcbY: 26.3,
-        pcbRotation: 90,
-        schX: 15.47,
-        schY: -3.96
+        pcbRotation: 90
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2809,7 +2858,7 @@ var Microcontroller_RP2040 = (_a) => {
         name: "C_3V3",
         capacitance: "10uF",
         footprint: "0603",
-        schSectionName: schSections.power,
+        schSectionName: schSections.power(name),
         schOrientation: "vertical",
         pcbX: -8.5,
         pcbY: 4.2
@@ -2821,12 +2870,12 @@ var Microcontroller_RP2040 = (_a) => {
         name: "C_CORE",
         capacitance: "1uF",
         footprint: "0402",
-        schSectionName: schSections.rp2040,
+        schSectionName: schSections.rp2040(name),
         schOrientation: "vertical",
         pcbX: 3.8,
         pcbY: -5.5,
-        schX: -2.8,
-        schY: -4.8
+        schX: -3.65,
+        schY: -3.7
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2835,12 +2884,10 @@ var Microcontroller_RP2040 = (_a) => {
         name: "C_USB",
         capacitance: "1uF",
         footprint: "0402",
-        schSectionName: schSections.usb,
+        schSectionName: schSections.usb(name),
         schOrientation: "vertical",
         pcbX: 9.8,
-        pcbY: 18.4,
-        schX: 16.11,
-        schY: -1.2
+        pcbY: 18.4
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2849,7 +2896,7 @@ var Microcontroller_RP2040 = (_a) => {
         name: "C_XIN",
         capacitance: "18pF",
         footprint: "0402",
-        schSectionName: schSections.clock,
+        schSectionName: schSections.clock(name),
         schOrientation: "vertical",
         pcbX: -8.4,
         pcbY: -11.8
@@ -2861,7 +2908,7 @@ var Microcontroller_RP2040 = (_a) => {
         name: "C_XOUT",
         capacitance: "18pF",
         footprint: "0402",
-        schSectionName: schSections.clock,
+        schSectionName: schSections.clock(name),
         schOrientation: "vertical",
         pcbX: -2.4,
         pcbY: -21
@@ -2873,11 +2920,13 @@ var Microcontroller_RP2040 = (_a) => {
         name: "L_AVDD",
         inductance: "600ohm@100MHz",
         footprint: "0603",
-        schSectionName: schSections.power,
+        schSectionName: schSections.power(name),
         pcbX: 8.5,
         pcbY: -1.8,
         supplierPartNumbers: { jlcpcb: ["C1002"] },
-        pcbRotation: 90
+        pcbRotation: 90,
+        schX: 6.8,
+        schY: -8.3
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2887,9 +2936,11 @@ var Microcontroller_RP2040 = (_a) => {
         footprintVariant: "pad",
         padShape: "circle",
         padDiameter: "1.1mm",
-        schSectionName: schSections.debug,
+        schSectionName: schSections.debug(name),
         pcbX: -6,
-        pcbY: -31
+        pcbY: -31,
+        schX: 6,
+        schY: -15.5
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2899,9 +2950,11 @@ var Microcontroller_RP2040 = (_a) => {
         footprintVariant: "pad",
         padShape: "circle",
         padDiameter: "1.1mm",
-        schSectionName: schSections.debug,
+        schSectionName: schSections.debug(name),
         pcbX: -2,
-        pcbY: -31
+        pcbY: -31,
+        schX: 7.5,
+        schY: -15.5
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2911,9 +2964,11 @@ var Microcontroller_RP2040 = (_a) => {
         footprintVariant: "pad",
         padShape: "circle",
         padDiameter: "1.1mm",
-        schSectionName: schSections.debug,
+        schSectionName: schSections.debug(name),
         pcbX: 2,
-        pcbY: -31
+        pcbY: -31,
+        schX: 9,
+        schY: -15.5
       }
     ),
     /* @__PURE__ */ jsx20(
@@ -2923,9 +2978,11 @@ var Microcontroller_RP2040 = (_a) => {
         footprintVariant: "pad",
         padShape: "circle",
         padDiameter: "1.1mm",
-        schSectionName: schSections.debug,
+        schSectionName: schSections.debug(name),
         pcbX: 6,
-        pcbY: -31
+        pcbY: -31,
+        schX: 10.5,
+        schY: -15.5
       }
     ),
     /* @__PURE__ */ jsx20(
